### DISCUSSION TITLE
AG Grid - JavaScript field evaluation - Nested fields

### DISCUSSION THREAD

**Discussion by bapowell** [elimintz/justpy#445] [Ideas]
Referring to [Tutorial-Grids-FieldEvaluation](https://justpy.io/grids_tutorial/field_evaluation/), this currently only handles fields at the top level of the options dictionary.

An improvement would be to handle nested fields that need to be JavaScript-evaluated.

An example would be formatting functions defined in columnTypes and/or columnDefs.

**Comment by elimintz** [comment]
Yes, in the new version (coming out in a couple of days), this is implemented. Every field that has `formatter` in its name, is evaluated as a javascript expression.

**Comment by WolfgangFahl** [comment]
formatter is currently only in a comment:
#TODO: Handle formatter functions, for example in dataLabels and others.
in chartcomponents

future idea or next milestone?

**Comment by elimintz** [comment]
I believe @rodja suggested separating issues with components from the core functionality and stability.  I think we should focus on the latter. AgGrid has so many features, it would be a big task to accommodate all of them

**Comment by brandonbettencourt** [comment]
@elimintz did that ever get incorporated in a release (that every field that has "formatter" in its name is evaluated as javascript)? I am currently trying to use the valueFormatter field but given that field is nested in grid options it is not working through the grid.evaluate call.
